src/server/pegasus_server_impl.cpp

```diff
@@ -2618,5 +2618,63 @@ void pegasus_server_impl::release_db()
     _db = nullptr;
 }
 
+void pegasus_server_impl::dwarn_write_request(dsn::message_ex *request,
+                                              uint64_t max_allowed_write_size)
+{
+
+    dsn::task_code rpc_code(request->rpc_code());
+    if (rpc_code == dsn::apps::RPC_RRDB_RRDB_PUT) {
+        auto put = put_rpc::auto_reply(request).request();
+        ::dsn::blob hash_key, sort_key;
+        pegasus_restore_key(put.key, hash_key, sort_key);
+        dwarn_replica("abnormal put: client_address = {}, hash_key = {}, sort_key = {}, put_bytes "
+                      "= {}, max_allowed_write_size = {}",
+                      request->header->from_address.to_std_string(),
+                      ::pegasus::utils::c_escape_string(hash_key).c_str(),
```

Comment:
dwarn_replica  is able to deal with std::string, so all `.c_str()` can be removed.